Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_controller.tsx

```diff
@@ -18,13 +32,55 @@ export class AccountDirectoryController extends
   constructor(props: Properties) {
     super(props);
     this.state = {
-      isLoaded: false
+      isLoaded: false, 
+      openedGroups: new Beam.Map<Beam.DirectoryEntry, AccountEntry[]>(),
+      filter: '',
+      filteredGroups: new Beam.Map<Beam.DirectoryEntry, AccountEntry[]>()
     };
   }
 
   public render(): JSX.Element {
     if(!this.state.isLoaded) {
       return <LoadingPage/>;
     }
+    <AccountDirectoryPage
+      displaySize={this.props.displaySize}
+      roles={this.props.roles}
+      groups={this.props.model.groups}
+      openedGroups={this.state.openedGroups}
+      filter={this.state.filter}
+      filteredGroups={this.state.filteredGroups}
+      onFilterChange={this.onFilterChange} 
+      onCardClick={this.onCardClick}/>
+  }
+
+  private async onCardClick(group: Beam.DirectoryEntry) {
+    if(this.state.openedGroups.get(group)) {
+      this.state.openedGroups.remove(group);
+    } else {
+      this.timerId = setTimeout(
+        async () => {

```

Comment:
Remove the timeout.  That might have been there for testing purposes but won't be needed any more.